docs/user-guide/kubectl-overview.md

```diff
@@ -33,160 +33,261 @@ Documentation for other releases can be found at
 
 # kubectl overview
 
+Use this overview of the `kubectl` command line interface to help you start running commands against Kubernetes clusters. We will quickly cover `kubectl` syntax, describe the command operations, and provide common examples. For details about each command, including all the supported options, see the [kubectl](kubectl/kubectl.md) reference documentation.
+
 **Table of Contents**
 
-- [kubectl overview](#kubectl-overview)
-  - [Overview](#overview)
-  - [Common Operations](#common-operations)
-  - [Kubectl Operations](#kubectl-operations)
-  - [Resource Types](#resource-types)
+ * [Syntax](#syntax)
+ * [Operations](#operations)
+ * [Resource types](#resource-types)
+ * [Output options](#output-options)
+ * [Examples: Common operations](#examples-common-operations)
+
+TODO: Auto-generate this file to ensure it's always in sync with any `kubectl` changes, see [#14177](https://github.com/kubernetes/kubernetes/pull/14177).
+
+## Syntax
+
+Use the following syntax to run `kubectl` commands from your terminal window:
+
+```
+kubectl [command] [TYPE] [NAME] [options]
```

Comment:
Why use `[options]` instead of `[flags]`? I used `[flags]` according to `kubectl-conventions.md` and `kubectl --help` output.